src/Database/Persist/Migration.hs

```diff
@@ -37,7 +37,7 @@ hasMigration :: MonadIO m => Persistent.Migration -> Persistent.SqlPersistT m Bo
 hasMigration = fmap (not . null) . Persistent.showMigration
 
 -- | Fails if the persistent library detects more migrations unaccounted for.
-checkMigration :: MonadIO m => Persistent.Migration -> Persistent.SqlPersistT m ()
+checkMigration :: (MonadIO m, MonadFail m) => Persistent.Migration -> Persistent.SqlPersistT m ()
```

Comment:
instead of adding monadfail constraints, change fail to error. this wont work if someone needs to use this library on ghc < 8.8 (because MonadFail is not in Prelude before then)